fedimint-bitcoind/src/bitcoincore.rs

```diff
@@ -100,6 +100,20 @@ impl IBitcoindRpc for BitcoinClient {
         Ok(height.map(|h| h as u64))
     }
 
+    async fn is_tx_in_block(
+        &self,
+        txid: &Txid,
+        block_hash: &BlockHash,
+        block_height: u64,
+    ) -> anyhow::Result<bool> {
+        let block_info = block_in_place(|| self.0.get_block_info(block_hash))?;
+        anyhow::ensure!(
+            block_info.height as u64 == block_height,
+            "Block height for block hash does not match expected height"
+        );
+        Ok(block_info.tx.contains(txid))
```

Comment:
This is slightly inefficient if we are waiting for multiple transactions since the block is being fetched multiple times. For now this isn't a problem and the alternative of just returning the transaction ids from the block wouldn't work with the electrum backend.